src/Swarm/Language/Syntax.hs

```diff
@@ -1027,3 +1030,13 @@ freeVarsV = freeVarsT . (\f -> \case TVar x -> TVar <$> f x; t -> pure t)
 -- | Apply a function to all free occurrences of a particular variable.
 mapFreeS :: Var -> (Syntax' ty -> Syntax' ty) -> Syntax' ty -> Syntax' ty
 mapFreeS x f = freeVarsS %~ (\t -> case t ^. sTerm of TVar y | y == x -> f t; _ -> t)
+
+-- | Transform the AST into a Tree datatype.
+-- Useful for pretty-printing (e.g. via "Data.Tree.drawTree").
+asTree :: Data a => Syntax' a -> Tree (Syntax' a)
+asTree = para Node
+
+-- | Each constructor is a assigned a value of 1, plus
+-- any recursive syntax it entails.
+measureAstSize :: Data a => Syntax' a -> Int
```

Comment:
Could we have a doctest that shows two solution code sizes?

I would like to see that this really measures the constructors that I hope it measures.

Reply:
Added several unit tests.